Improve proxy security (#100) (#134)
hplush/slowreader: proxy/README.md

HTTP-server to proxy all RSS fetching request from web client.

User could use it to bypass censorship or to try web client before they install upcoming extension (to bypass CORS limit of web app).

> ## Usage

Author: Fixed `README.md`

### Start Proxy

```sh
pnpm start
# // Proxy server running on port 5284
```

> ## Additional Info:

Reviewer: Title should not contain `:`

Author: Just fixed `README.md`

HTTP-server to proxy all RSS fetching request from web client.

User could use it to bypass censorship or to try web client before they install upcoming extension (to bypass CORS limit of web app).

## Scripts

### Start Proxy

```sh
pnpm start
# // Proxy server running on port 5284
```

## Abuse Protection

- Proxy allows only GET requests

> - Proxy do not allow requests to reserved ip addresses like `127.0.0.1`

Reviewer: ```suggestion
- Proxy do not allow requests to in-cloud IP addresses like `127.0.0.1`
```

> - Proxy allows only http or https

Reviewer: ```suggestion
- Proxy allows only HTTP or HTTPS protocols
```

- Proxy removes cookie headers

## Test Strategy

> Proxy is tested using e2e testing. To write tests `initTestHttpServer` should be used

Reviewer: ```suggestion
To test proxy we emulate the real HTTP servers (end-to-end testing).
```

Docs are for juniors.

HTTP-server to proxy all RSS fetching request from web client.

User could use it to bypass censorship or to try web client before they install upcoming extension (to bypass CORS limit of web app).

## Usage

### Start proxy

> ```shell

Reviewer: ```suggestion
```sh
```

pnpm start
# // Proxy server running on port 5284
```

### Make a request

```shell

> curl localhost:5284/https://hplush.dev

Reviewer: Let’s not document the API 😏

Author: Alright :-)

Author: Done!
